added some dao + completed import.sql

BenMorant opened the pull request:
added some dao + completed import.sql

Commit:
formatted all my code with google checkstyle setting xml, available here : https://raw.githubusercontent.com/google/styleguide/gh-pages/intellij-java-google-style.xml.

Commit:
formatted all the code with the plugin google-java-format

https://plugins.jetbrains.com/plugin/8527-google-java-format

Commit:
formatted all my code with google checkstyle setting xml, available here : https://raw.githubusercontent.com/google/styleguide/gh-pages/intellij-java-google-style.xml.

Commit:
Created item and peopleitem entities, + completed import.sql accordingly .

BenMorant committed:
Created item and peopleitem dao and business

Commit:
completed import.sql

fixed “Error executing DDL ”alter table xxx drop foreign key yyy“ via JDBC Statement” by modifying spring.jpa.hibernate.ddl-auto , from create-drop to update .

Commit:
added gradle declaration 6.5 to build.gradle .

BenMorant committed:
modified field weight => slot in item + modified import.sql

Commit:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

BenMorant committed:
#12 added spotbugs in build.gradle

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

BenMorant committed:
added assertj-core in build.gradle

xavierhardy committed:
#12 fixed SpotBugs Gradle plugin and warning

BenMorant committed:
added assertj-core in build.gradle

BenMorant committed:
modified spring.jpa.hibernate.ddl-auto=update to create .

BenMorant committed:
Merge remote-tracking branch 'origin/bmorant' into bmorant

BenMorant committed:
removed association class PeopleItem , its DAO and service.
Created Inventory and made the links + completed import.sql accordingly.

BenMorant committed:
reformat all the code .

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

 Conflicts:
	src/main/java/com/benmorant/game/holycraap/HolycraapApplication.java

BenMorant committed:
reformat all the code .

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

 Conflicts:
	.github/workflows/build.yml
	src/main/java/com/benmorant/game/holycraap/model/PeopleController.java
	src/main/java/com/benmorant/game/holycraap/model/business/PeopleServiceImpl.java
	src/main/java/com/benmorant/game/holycraap/model/dao/PeopleDao.java
	src/main/java/com/benmorant/game/holycraap/model/entity/People.java
	src/test/java/com/benmorant/game/holycraap/HolycraapApplicationTests.java

BenMorant committed:
reformated all the code

BenMorant committed:
deleted PeopleController

BenMorant committed:
deleted PeopleController

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

 Conflicts:
	src/main/java/com/benmorant/game/holycraap/HolycraapApplication.java
	src/main/java/com/benmorant/game/holycraap/model/PeopleController.java

BenMorant committed:
Removed PeopleController.java and PeopleControllerTest.java (cf. issue #25)
added Serializable to the class Inventory .

BenMorant committed:
Reformated again

BenMorant commented:
Finally all checks passed ! 
Champagne !

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

 Conflicts:
	src/test/java/com/benmorant/game/holycraap/controller/PeopleControllerTest.java

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

 Conflicts:
	src/main/resources/application.properties

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

BenMorant committed:
changed some annotations + import.sql .

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

BenMorant committed:
modified relation between People and Item, deleted Inventory, modified import.sql consequently

xavierhardy commented on build.gradle:
```diff
@@ -45,6 +45,11 @@ repositories {
+wrapper {
+    gradleVersion = '6.5'
+    distributionType = Wrapper.DistributionType.BIN
+}
```
This shouldn't be necessary if we don't use Gradle 6.5 features (same thing for forcing the use of JDK 11).

BenMorant replied:
It might be necessary IMHO. 
I dunno why, but when I start a gradle project with IntelliJ IDEA, gradle's version is 6.1 by default and I don't know how to change that (and 'if' I can change that btw...). 
But Spring Boot needs an ulterior gradle version to work.  (>6.3 if I remember correctly)
Concerning jdk 11, we might use some specific features... someday.

Anyway, I'll remove that from the file, but keep the stuff in mind.

xavierhardy commented on src/main/resources/import.sql:
```diff
@@ -2,3 +2,9 @@
+/* Item */
+insert into `item` (name, price, slot, people_id) values ('Dent de dragon', '100', '1', '1');
+insert into `item` (name, price, slot, people_id) values ('salicorne', '2', '1', '2');
+insert into `item` (name, price, slot, people_id) values ('gousse de mandragore', '5', '1', '3');
```
Use integers (no quote `'`) when possible.

BenMorant replied:
agreed and modified.

BenMorant committed:
#7 modified build.gradle so it doesn't force the use of gradle 6.5 and Java 11.

BenMorant committed:
#7 modified import.sql so no '' when numbers + now SQL REQUESTS are in cap'

BenMorant committed:
Merge branch 'master' of https://github.com/BenMorant/holycraap into bmorant

BenMorant committed:
After pulling master, I had build errors relative to
java.lang.AssertionError:
Expecting ArrayList:
 <[People(id=1, name=Loup Ardent, currentHp=12, hpMax=12, currentMp=6, mpMax=6, inventory=[Item(id=1, name=Dent de dragon, slot=1, price=100.0)]),
    People(id=2, name=Hardy Xav, currentHp=1000, hpMax=1000, currentMp=1000, mpMax=1000, inventory=[Item(id=2, name=salicorne, slot=1, price=2.0)]),
    People(id=3, name=Conan le Barbare, currentHp=14, hpMax=18, currentMp=4, mpMax=4, inventory=[Item(id=3, name=gousse de mandragore, slot=1, price=5.0)]),
    People(id=4, name=somename, currentHp=10, hpMax=11, currentMp=12, mpMax=13, inventory=[])]>
to contain:
 <[People(id=4, name=somename, currentHp=10, hpMax=11, currentMp=12, mpMax=13, inventory=null)]>
but could not find the following element(s):
 <[People(id=4, name=somename, currentHp=10, hpMax=11, currentMp=12, mpMax=13, inventory=null)]>

 => I set up an item, saved it, added it to the inventory... Now it's shitty code, but it works.

xavierhardy commented on src/test/java/com/benmorant/game/holycraap/controller/PeopleControllerTest.java:
```diff
@@ -54,6 +59,16 @@ public void shouldReturnAllPeople() throws Exception {
     final int mpMax = 13;
+    final Item item = new Item();
+    item.setId(null);
+    item.setName(null);
```
You shouldn't need to set it when creating the item.

xavierhardy reviewed (approved):
Overall, this looks good to me.

BenMorant commented:
finally ! :)
